engine/crates/engine-v2/schema/src/lib.rs

```diff
@@ -397,14 +398,28 @@ impl Schema {
     }
 }
 
-#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
-pub struct Header {
-    pub name: StringId,
-    pub value: HeaderValue,
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub enum NameOrPattern {
+    /// A regex pattern matching multiple headers.
+    #[serde(with = "serde_regex", rename = "pattern")]
+    Pattern(Regex),
+    /// A static single name.
+    #[serde(rename = "name")]
```

Comment:
won't work with `postcard`. Sure the regex stuff will?

Reply:
Regex stuff does work, the renames are useless and I will remove them here.